shared/src/code.rs

```diff
@@ -0,0 +1,17 @@
+use crate::Web3;
+use anyhow::Result;
+use ethcontract::H160;
+
+#[mockall::automock]
+#[async_trait::async_trait]
+pub trait CodeFetching: Send + Sync {
```

Comment:
nanonit: Maybe all these helper traits could be combined in a web3 module or the like?

Reply:
As far as I can tell, this is the only one for now:
```
$ rg 'for Web3'
orderbook/src/account_balances.rs
140:impl BalanceFetching for Web3BalanceFetcher {

shared/src/code.rs
13:impl CodeFetching for Web3 {
```

I moved it to a `web3_traits` module that we can add to over time.